moveit_planners/trajopt/src/trajopt_interface.cpp

```diff
@@ -0,0 +1,434 @@
+  {
+    CartPoseTermInfoPtr cart_goal_pos(new CartPoseTermInfo);
+
+    // TODO: Feed cart_goal_pos with request information and the needed param to the setup.yaml file
+    // TODO: Multiple Cartesian constraints
+
+    // Add the constraint to problem_info
+    problem_info.cnt_infos.push_back(cart_goal_pos);
+  }
+  else if (req.goal_constraints[0].position_constraints.empty() &&
+           !req.goal_constraints[0].orientation_constraints.empty())
+  {
+    ROS_ERROR_STREAM_NAMED("trajopt_planner", "position constraint is not defined");
+    res.error_code.val = moveit_msgs::MoveItErrorCodes::INVALID_GOAL_CONSTRAINTS;
+    return false;
+  }
+  else if (!req.goal_constraints[0].orientation_constraints.empty() &&
+           req.goal_constraints[0].orientation_constraints.empty())
+  {
+    ROS_ERROR_STREAM_NAMED("trajopt_planner", "orientation constraint is not defined");
+    res.error_code.val = moveit_msgs::MoveItErrorCodes::INVALID_GOAL_CONSTRAINTS;
+    return false;
+  }
+
+  ROS_INFO(" ======================================= Constraints from request goal_constraints");
+  for (auto goal_cnt : req.goal_constraints)
+  {
+    JointPoseTermInfoPtr joint_pos_term(new JointPoseTermInfo);
+    setJointPoseTermInfoParams(joint_pos_term, goal_cnt.name);
+
+    trajopt::DblVec joint_goal_constraints;
+    for (const moveit_msgs::JointConstraint& joint_goal_constraint : goal_cnt.joint_constraints)
+    {
+      joint_goal_constraints.push_back(joint_goal_constraint.position);
+    }
+    joint_pos_term->targets = joint_goal_constraints;
+    problem_info.cnt_infos.push_back(joint_pos_term);
+  }
+
+  ROS_INFO(" ======================================= Constraints from request start_state");
+  // add the start pos from request as a constraint
+  JointPoseTermInfoPtr joint_start_pos(new JointPoseTermInfo);
+  trajopt::DblVec joint_start_constraints;
+  for (auto pos : req.start_state.joint_state.position)
+  {
+    joint_start_constraints.push_back(pos);
+  }
+  joint_start_pos->targets = joint_start_constraints;
+  setJointPoseTermInfoParams(joint_start_pos, "start_pos");
+  problem_info.cnt_infos.push_back(joint_start_pos);
+
+  ROS_INFO(" ======================================= Velocity Constraints, hard-coded");
+  // TODO: should be defined by user, its parametes should be added to setup.yaml
+  JointVelTermInfoPtr joint_vel(new JointVelTermInfo);
+
+  joint_vel->coeffs = std::vector<double>(7, 5.0);
+  joint_vel->targets = std::vector<double>(7, 0.0);
+  joint_vel->first_step = 0;
+  joint_vel->last_step = problem_info.basic_info.n_steps - 1;
+  joint_vel->name = "joint_vel";
+  joint_vel->term_type = trajopt_interface::TT_COST;
+  problem_info.cost_infos.push_back(joint_vel);
+
+  ROS_INFO(" ======================================= Visibility Constraints");
+  if (!req.goal_constraints[0].visibility_constraints.empty())
+  {
+    // TODO: Add visibility constraint
+  }
+
+  ROS_INFO(" ======================================= DEBUG");
```

Comment:
If these are debug prints, shouldn't they be `ROS_DEBUG`?

Reply:
yes. I change them to ROS_DEBUG